gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/resources/scripts/3.17.0/api-keys-cleanup.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * This script should be run before upgrading in order to avoid any issue
+ * linked to https://github.com/spring-projects/spring-data-mongodb/issues/2350$
+ * while running the ApiKeySubscriptionUpgrader
+ */
+
+// Override this variable if you use prefix
+const prefix = '';
+
+function unsetUndefined(doc) {
+    return Object.keys(doc).some((k, i) => {
+        if (doc[k] === undefined) {
+            print('found an undefined field on API key', doc._id, k, 'will be unset');
+            delete doc[k];
+            return true;
+        }
+    });
+}
+
+db.getCollection(`${prefix}keys`).find({}).forEach(doc => {
+    const changed = unsetUndefined(doc);
+    if (changed) {
+        db.getCollection(`${prefix}keys`).save(doc);
```

Comment:
Should we use `replaceOne` instead of `save`, which is deprecated in recent mongoDb versions ?
(see https://www.mongodb.com/docs/v4.4/reference/method/db.collection.save)

Reply:
is it documentDB compliant ?

Reply:
Has to be checked but I think so.

Some DocumentDb users had troubles using deprecated `save` method in their mongo scripts (see zendesk #2851).
Since that, we decided to not use it nomore (https://github.com/gravitee-io/issues/issues/6260).

Reply:
yes I think it is supported by documentDB as there is no reference what so ever in the compatibility matrix.

@marcambier done